cupyx/fallback_mode/fallback.py

```diff
@@ -141,16 +169,64 @@ def __new__(cls, *args, **kwargs):
             return object.__new__(cls)
 
         cupy_ndarray_init = cp.ndarray(*args, **kwargs)
-        return cls(_stored=cupy_ndarray_init)
+        return cls(_stored=cupy_ndarray_init, _class=cp.ndarray)
 
     def __init__(self, *args, **kwargs):
+        """
+        Args:
+            _stored (None, ndarray): If _stored is None, object is not
+            initialized. Otherwise, _stored (ndarray) would be set to
+            _cupy_array and/or _numpy_array depending upon _class.
+            _class (ndarray type): If _class is `cp.ndarray`, _stored is
+            set as _cupy_array and _numpy_array. Otherwise, _stored is set
+            as only _numpy_array.
+            Intended values for _class are `np.ndarray`, `np.ma.MaskedArray`,
+            `np.matrix`, `np.chararray`, `np.recarray`.
+
+        Attributes:
+            _cupy_array (cp.ndarray): ndarray fully compatible with CuPy.
+            This will be always set to a ndarray in GPU.
+            _numpy_array (np.ndarray and variants): ndarray not supported by
+            CuPy. Such as np.ndarray(where dtype is not in '?bhilqBHILQefdFD')
+            and it's variants. This will be always set to a ndarray in CPU.
+            _class (ndarray type): If _class is `cp.ndarray`, data of array
```

Comment:
Is `_class` always a ndarray type? "NumPy's class or cupy.ndarray" seems more appropriate.

Reply:
Anyway, this class is needed? How about adding the following property?
```py
@property
def _class(self):
    return type(self._cupy_array if self._cupy_array is not None else self._numpy_array)
```

Reply:
How about this comment?

Reply:
`_class` is needed after fallback. During fallback we may need to update `_cupy_array` from `_numpy_array`, depending on cupy compatibility of array. Say if `type(array) is np.ndarray`, then we're not sure if it is compatible with cupy or not. We will then need to check `if array.dtype.kind in '?bhilqBHILQefdFD'`. That's what `_class` is made from during creation of `ndarray`. Checking dtype or using `_class` is same thing. Whichever you prefer is fine.

Reply:
If so, how about fixing it to boolean `_enable_cupy` for example?

Reply:
Yeah, slight modifications will be needed in one or two places like L293 (L307), but using boolean is cleaner. How about `_supports_cupy`?

Reply:
Done